### PR TITLE
Adding cfn test workflow

### PR DESCRIPTION
**Pull request by govardhanpagidi**
## Description

Added cfn test workflow
Refactored Project and tested  CFN TEST using cfn-test-helper.sh

Link to any related issue(s): 

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run `make fmt` and formatted my code

### CODE REVIEW

File: .github/workflows/cfn-test.yaml
Comment (andreaangiolillo) [inline .github/workflows/cfn-test.yaml:32]:
```suggestion
            CFN_FLAGS="--verbose --set-default --region ${AWS_DEFAULT_REGION}" ./cfn-testing-helper.sh
```

File: cfn-resources/project/test/cfn-test-create-inputs.sh
Comment (andreaangiolillo) [inline cfn-resources/project/test/cfn-test-create-inputs.sh:24]:
```suggestion
```

**Review by andreaangiolillo — COMMENTED** [review]
a left a few comments. No need to wait for my approval once my comments have been addressed. Thanks!

**Review by Zuhairahmed — APPROVED** [review]
LGTM

File: cfn-resources/project/cmd/resource/resource.go
Comment (govardhanpagidi) [reply cfn-resources/project/cmd/resource/resource.go:144]: event will have the error response already, if we send err response from handler it always return GenericException rather Actual Error Response.

Comment (andreaangiolillo) [reply cfn-resources/project/cmd/resource/resource.go:144]: Thanks for the clarification

**Review by ParthasarathyVaradhan — APPROVED** [review]
LGTM

**Pull request merged**
